tangelo/toolboxes/operators/operators.py

```diff
@@ -27,10 +28,54 @@
 
 
 class FermionOperator(openfermion.FermionOperator):
-    """Currently, this class is coming from openfermion. Can be later on be
-    replaced by our own implementation.
+    """Add n_spinorbitals, n_electrons, and spin parameters to the openfermion.FermionOperator class
     """
 
+    def __init__(self, term=None, coefficient=1., n_spinorbitals=None, n_electrons=None, spin=None):
+        super(FermionOperator, self).__init__(term, coefficient)
+
+        self.n_spinorbitals = n_spinorbitals
+        self.n_electrons = n_electrons
+        self.spin = spin
+
+    def __iadd__(self, other):
+        # Raise error if attributes are not the same across Operators.
+        if (self.n_spinorbitals is not None and other.n_spinorbitals is not None)\
```

Comment:
I think this the right way of doing this check, as opposed to what QubitHamiltonian is doing